test/correctness/simd_op_check.cpp

```diff
@@ -1737,11 +1737,12 @@ class SimdOpCheck : public SimdOpCheckTest {
                 check("i64x2.mul", 2 * w, i64_1 * i64_2);
 
                 // Integer dot product (16 -> 32)
-                // TODO(https://github.com/halide/Halide/issues/5130): NOT BEING GENERATED AT TRUNK
-                // {
-                //     RDom r(0, 4);
-                //     check("i32x4.dot_i16x8_s", 2 * w, sum(i32(in_i16(x * 4 + r)) * in_i16(x * 4 + r + 32)));
-                // }
+                for (int f : {4, 8}) {
```

Comment:
I think you should test 2 here, since that's really what the instruction is targeting.

Reply:
done